include/boost/url/detail/impl/remove_dot_segments.ipp

```diff
@@ -533,6 +558,28 @@ normalized_path_compare(
     return 1;
 }
 
+void
+normalized_path_digest(
+    string_view s,
+    bool remove_unmatched,
+    std::size_t& hash,
+    std::size_t prime) noexcept
+{
+    string_view child;
+    std::size_t level = 0;
+    do
+    {
+        pop_last_segment(
+            s, child, level, remove_unmatched);
+        while (!child.empty())
+        {
+            char c = path_pop_back(child);
+            hash = (c ^ hash) * prime;
```

Comment:
This logic came from boost.json string. I don't know if there's anything we have to adjust here.

Reply:
hmmm `c ^ hash` is not producing much information. Ideally `c` would be the same number of bits as `hash`.

Reply:
it turns out boost.json is wrong, see https://en.wikipedia.org/wiki/Fowler%E2%80%93Noll%E2%80%93Vo_hash_function